### PR TITLE
fix(cli): allowlist reserved synthetic values and GitHub IDs in PII scanner

### LINKED ISSUE #1343: scorer: pii-audit + wholesale-accept-no-fixes should exempt RFC 2606 / RFC 6761 reserved domains
## Summary

`printing-press pii-audit` (the email detector) flags any `@example.com` address as an email PII finding. The `wholesale-accept-no-fixes` gate then fires when those findings are accepted (correctly, with `category: synthetic_placeholder`) without an accompanying source-side rename. For CLI narrative recipes that legitimately need example emails (the RFC 2606 use case the IANA reservation was created for), there is no way to "source-fix" the recipes without scaffolding the rendered docs with a `PII_EMAIL_EXAMPLE` sentinel that polish's own playbook explicitly forbids ("Never add scaffolding to satisfy the scorer").

Net effect: a Grade-A run promotes to `hold` purely because its rendered recipes use the standards-reserved synthetic email domain that those recipes are *supposed* to use.

## Evidence

Run `microsoft-graph` mail+calendar (`/Users/hwt/printing-press/manuscripts/microsoft-graph/20260514-000037/`). Shipcheck PASS 6/6 legs, scorecard 84/100 Grade A. After accepting 15 RFC 2606 `@example.com` findings as `synthetic_placeholder` with per-finding `evidence_context`, `printing-press lock promote` failed:

```
all 13 finding(s) accepted with zero source fixes from baseline — agent stamped accept without fixing real PII
```

Polish's own remaining-issues block:

> PII gate failure 'wholesale-accept-no-fixes' (1 gate failure, 0 pending): 15 RFC-2606 @example.com accepts are correct synthetic_placeholder values; clearing the gate would require inserting PII_EMAIL_EXAMPLE sentinels into rendered docs, which is scaffolding-to-satisfy-the-gate. Generator-level fix (recipe-density reduction or sentinel policy) is retro scope.

## Cross-API check (Phase 3 Step B)

Three concrete candidates with evidence:

- **Notion** (catalog `productivity:`): `developers.notion.com/reference/post-page` uses `assignee@example.com` consistently. Any Notion CLI's `pages create` recipe hits this.
- **Linear**: Linear's official curl examples use `@example.com` for issue assignees. Any Linear CLI recipe creating or assigning issues will surface this.
- **Stripe** (`stripe.com/docs/api/customers/create`): uses `jenny.rosen@example.com` throughout. Any Stripe CLI customer-create recipe will surface this.

Conservative estimate: 60-70% of catalog will hit this whenever recipes include recipient/assignee emails.

## Counter-check

Would the fix hurt APIs that don't have email recipes? **No.** The fix matches purely on IANA-reserved (`example.com`, `example.org`, `example.net`) and RFC 6761-reserved (`.test`, `.example`, `.invalid`, `.localhost`) domain strings. Real PII never uses these domains; the reservation exists for this exact use case.

## Proposed fix (preferred to fallback)

1. **Detector exemption (preferred):** In `pii-audit`'s email detector, skip findings whose right-hand-side domain matches:
   - Exact: `example.com`, `example.org`, `example.net`
   - Suffix: `.test`, `.example`, `.invalid`, `.localhost`
   
   Treat as not-a-finding rather than auto-accept. The user never sees them; no entry in the ledger; nothing for the wholesale-accept gate to count.

2. **Gate exemption (fallback if the detector is kept conservative):** In the wholesale-accept-no-fixes gate, exclude `category: synthetic_placeholder` accepts whose `matched_span` ends in a reserved-domain string from the source-fix-required count.

3. **Generator-side sentinel (last resort):** Renderer emits a `PII_EMAIL_EXAMPLE` token. Bad: reader-facing recipes lose readability; polish's own "never scaffold" rule argues against this.

## Test shape

Positive (detector path):
```
echo 'See alice@example.com and bob@example.org' > /tmp/fixture.md
printing-press pii-audit /tmp/  # expects: 0 email findings
```

Positive (real PII still flagged):
```
echo 'See alice@example.com and alice@realdomain.com' > /tmp/fixture.md
printing-press pii-audit /tmp/  # expects: exactly 1 email finding (the realdomain.com one)
```

Negative (real-PII wholesale-accept still caught):
```
# README with 13 real-domain emails, all accepted as synthetic_placeholder
printing-press lock promote --cli foo --dir /tmp/foo  # MUST still fire wholesale-accept-no-fixes
```

Negative (reserved-domain doesn't bypass gate for real findings):
```
# README with 13 reserved-domain emails (auto-suppressed) + 5 real-domain emails accepted
printing-press lock promote --cli foo --dir /tmp/foo  # MUST fire wholesale-accept-no-fixes on the 5 real
```

## Related

- Polish skill playbook: `references/pii-polish.md` "Never add scaffolding to satisfy the scorer" rule, which is correct and should not change.

## Related issues

None in current open scan. First sighting of this pattern in the retro tracker.

---

🤖 Generated from Printing Press retro `microsoft-graph` run `20260514-000037` by Claude Opus 4.7 (1M context).

Full retro doc: `/Users/hwt/printing-press/manuscripts/microsoft-graph/20260514-000037/proofs/20260514-004933-retro-microsoft-graph-mail-pp-cli.md`

### LINKED ISSUE #1477: scorer: publish-package PII scanner blocks on GitHub comment IDs in archived retro/manuscript proofs
## Symptom

`printing-press publish package` aborts with `customer PII detected in staged package: <path> phone-us [hash] <10-digit-id>` when the staging tree contains a 10-digit GitHub comment ID inside an archived retro doc, manuscript, or any included artifact.

## Reproducer

The retro doc for any CLI that references a prior Greptile inline review will contain comment IDs like `3249672558`. Publish package walks `.manuscripts/<run-id>/proofs/*.md`, runs the PII scanner, and blocks on the first 10-digit run that matches NANP shape (area code 2-9, exchange 2-9):

```
$ printing-press publish package --dir ~/printing-press/library/<api> --category <cat> --target /tmp/staging --json
customer PII detected in staged package:
  .manuscripts/<run-id>/proofs/<retro>.md:26 phone-us [f6631e25ab40] 3249672558
  .manuscripts/<run-id>/proofs/<retro>.md:43 phone-us [b3c58f79d9c6] 3249672648
```

The same issue surfaces on any 10-digit run starting with 2-9 — review-comment IDs, PR commit-id-shaped numbers, etc. Workflow run IDs (11 digits) and issue numbers (4-5 digits) are safe.

## Root cause

`internal/artifacts/pii.go:118` — the NANP phone regex correctly identifies real US phone numbers, but a 10-digit run starting with 2-9 matches by shape alone:

```go
{
    kind: PIIKindPhoneUS,
    pattern: regexp.MustCompile(`\b(?:\+?1[\s.\-]?)?\(?[2-9]\d{2}\)?[\s.\-]?[2-9]\d{2}[\s.\-]?\d{4}\b`),
},
```

GitHub comment IDs are well-formed 10-digit numbers in the matching range. The scanner has no allowlist mechanism, no per-file ignore, no pragma — every match aborts the package.

## Why this matters

Retro docs that cite prior PR review work as evidence (the documented retro pattern in `skills/printing-press-retro/SKILL.md` Phase 5) include comment IDs by convention. Once a CLI ships its first retro to manuscripts, every subsequent publish package run aborts.

The friction is silent at write-time and loud at publish-time. The agent has no way to know they've created a publish-blocking artifact until they hit it.

## Proposed fix

Three viable approaches, listed by impact:

**A. Allowlist GitHub-shaped reference IDs in archived markdown.**
Add a positive-context check: if the matching digit run is preceded within ~30 chars by `comment id`, `discussion_r`, `commit `, `pull/`, `#`, or similar GitHub-context tokens, skip the finding. Narrow targeting; covers the documented use case without weakening real-PII detection.

**B. `// pp:allow-pii <kind>` line pragma.**
Add an inline pragma agents can place on a line that legitimately contains a phone-shaped ID (or other PII shape) with a documented reason. Mirrors the `// pp:client-call`, `// pp:novel-static-reference`, etc. existing pragma pattern documented in `AGENTS.md` "Anti-reimplementation" section. More general but adds an authoring step.

**C. Skip `.manuscripts/**/*.md` in the scan path.**
The scan exists to catch live PII in shipped CLI source. Archived markdown proofs aren't shipped to end users — they're publish-time artifacts for maintainers. Skipping the path entirely loses defense-in-depth for the proof contents but recognizes the actual shape of what `.manuscripts/` carries.

(A) is the smallest fix and handles the most common false-positive shape without removing scan coverage. (C) is the most invasive but recognizes the manuscript-shape correctly. (B) is the most general but pushes work to the agent.

## Workaround in the meantime

Format GitHub comment IDs in retro docs and manuscripts as `discussion_r<id>` (matches GitHub's actual URL-fragment convention). The `r` prefix breaks the `\b` word boundary the regex requires.

I've added this workaround to my session memory under `feedback_pp_pii_scanner_phone_false_positive.md` so future retros use the safe format from the start. A future agent without that memory will trip the same scanner.

## Acceptance

- positive: a retro doc citing `comment id discussion_r3249672558` and `comment id 3249672558` (within a recognized GitHub-context phrase) both pass the scanner. Real US phones in API response bodies (e.g., `"phone": "(415) 555-0123"`) still fail.
- negative: existing `internal/artifacts/pii_test.go` cases continue to pass.

## Evidence

- Local reproducer: `/Users/<...>/printing-press/library/<api>/.manuscripts/<run-id>/proofs/<retro>.md` containing `comment id 3249672558` (where 3249672558 is the actual review-comment ID for [mvanhorn/printing-press-library#585 review-comment `discussion_r3249672558`](https://github.com/mvanhorn/printing-press-library/pull/585#discussion_r3249672558)).
- Aborted package run during retro filing for the youtube CLI on 2026-05-15.

## Complexity

Small for option (A) or (C); medium for option (B) (requires pragma scanner integration, plus updating the retro/publish skills to document the pragma).

---

_Filed as a follow-up from the youtube CLI run (`20260515-050804`) retro and publish flow._

### LINKED ISSUE #1589: Publish-package PII guard tier-2 has no allowlist for RFC-reserved synthetic values (example.com, NANP 555-01XX)
> Filed by `/printing-press-retro` from a run of **emailoctopus**
> (scorecard 85/100, 100% verify pass).

## Summary

The publish-package PII guard's tier-2 pattern scanner (`email`, `phone-us`) has no allowlist for RFC-reserved synthetic values. RFC 2606 reserves `example.com`/`example.org`/`example.net` precisely so they don't collide with real domains; NANP reserves the `555-01XX` phone range as fictional. Live-dogfood routinely captures these in `output_sample` payloads when the API echoes back test inputs, and the guard then blocks publish with false positives. Current workaround is to delete the offending manuscript file or substitute non-pattern-matching tokens (which the guard catches too).

## Where to look

- **Component:** scorer (publish-package PII scanner)
- **Likely area:** `internal/publish/` — the tier-2 PII pattern scanner that runs during `printing-press publish package`. Per the main SKILL's `references/secret-protection.md`, tier-2 is the email/phone pattern layer that warns on generic matches.
- **Triggered when:** Any publish where dogfood captured a synthetic placeholder value in proofs

## What we observed

EmailOctopus publish-package fail on the staged tree (after the API key value was already auto-scrubbed):

```
customer PII detected in staged package:
  .manuscripts/20260517-015019/proofs/dogfood-full.json:890 email [72710f068529] user@example.com
  .manuscripts/20260517-015019/proofs/dogfood-full.json:930 email [9ccb66b9cdf3] user@example.com
  .manuscripts/20260517-015019/proofs/dogfood-full.json:595 phone-us [0a135d028936] 2087741623
  .manuscripts/20260517-015019/proofs/dogfood-full.json:605 phone-us [da8e2b2127de] 2087741623
```

Two distinct PII classes mixed in the same failure:

1. **Real PII** (the user's actual phone `2087741623`, the user's actual email `trevin@trevinchow.com`, a third-party subscriber's email `alib.seattle@gmail.com`) — correctly flagged; needs scrubbing.
2. **RFC-reserved synthetic values** (`user@example.com`) — false-positive; these are intentional placeholders the API echoed back from dogfood test inputs.

After deleting `dogfood-full.json` entirely and scrubbing real PII to `[REDACTED]`, attempted to substitute the synthetics to `subscriber@example.com` / `5555550100` / `printer@example.com` — guard caught those too because they still match the email/phone regex.

## Suspected root cause

Tier-2 scanner uses generic regex patterns for email (`[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,}`) and US phone (`\d{3}[-.]?\d{3}[-.]?\d{4}`) without an allowlist pass for RFC-reserved synthetic values.

## Suggested direction

Add a suffix-match allowlist for emails and prefix/range-match for US phones, applied **before** each tier-2 finding is recorded:

```go
// Email allowlist: RFC 2606 reserved + common test domains
emailAllowlistSuffixes := []string{
    ".example.com", ".example.org", ".example.net",
    ".test", ".localhost", ".invalid",
}
// US phone allowlist: NANP fictional 555-01XX range
phoneAllowlistRegex := regexp.MustCompile(`^555-?01\d{2}$`)

// Literal placeholder tokens (case-sensitive)
literalAllowlist := []string{"<redacted>", "[REDACTED]", "xxx@xxx.com"}
```

Allowlist entries match domain suffix (for emails) and the documented fictional prefix (for phones). The vendor-prefix tier-1 scanner (`sk_live_`, `ghp_`, etc.) stays unchanged — real-looking patterns are still caught.

## Acceptance criteria

- positive: Publish-package a CLI whose proofs contain `user@example.com`, `anything@example.org`, `5555550100`, `555-0123` variants; assert no tier-2 PII findings on those values.
- negative: Publish-package a CLI whose proofs contain a real-shaped `customer@gmail.com` or `2087741623`; assert the values still trip the guard.

## Frequency

every publish where dogfood captured a placeholder (will become more common as dogfood-full.json archiving becomes standard in manuscripts)

## Complexity

small

## Scope boundary

Only the tier-2 allowlist. Don't relax tier-1 (vendor-prefix tokens like `sk_live_*`). Don't add domain allowlisting beyond the RFC-reserved suffixes — broader allowlists (e.g., wildcarding `.com`) would create real false negatives.

## Dependencies

None.

## Related issues

- #1480 — `scorer: pii-audit false-positives on test-fixture addresses and .omc/ tooling workspace` (`related-area`) — adjacent class of tier-2 false positives, different specific allowlist class.
- #1477 — `scorer: publish-package PII scanner blocks on GitHub comment IDs in archived retro/manuscript proofs` (`related-area`) — same scanner, different false-positive pattern. Could batch with this WU under a single "scanner allowlist hardening" effort.

## Artifacts

| Artifact | Link |
|----------|------|
| Retro document | https://files.catbox.moe/us6uyq.md |
| Manuscripts (research + proofs) | https://files.catbox.moe/bt5ksb.zip |
| Generated CLI source code | https://files.catbox.moe/jx7pxm.zip |

---

*Generated by `/printing-press-retro` · [CLI Printing Press](https://github.com/mvanhorn/cli-printing-press)*

### PR DESCRIPTION
**Pull request by vinnyp**
## Intent

Issue: Closes #1589, Closes #1343, Closes #1477

Three issues, one root cause: the PII scanner (`internal/artifacts/pii.go`, shared by `pii-audit`, `publish package`, and the polish gates) had no allowlist for values that are reserved-by-standard or context-disambiguated. So it false-positived on:
- RFC 2606/6761 reserved synthetic emails (`user@example.com`) and phones (NANP `555-01XX`) that APIs echo back in dogfood output (#1589, #1343) — blocking publish or forcing polish churn that the "never scaffold to satisfy the gate" rule forbids resolving cleanly.
- 10-digit GitHub comment/commit/PR IDs that match the NANP phone shape and appear in archived retro/manuscript proofs (#1477) — every publish after the first retro aborted.

## Approach

Add narrow allowlists at the single shared chokepoint, `isSyntheticPIIPlaceholder` (called by both detection in `scanPIIFileWithRel` and redaction in `redactPIIPatterns`, so they can't drift):
- **Email:** skip `example.com`/`.org`/`.net` (and their subdomains — RFC 2606 reserves the whole tree) and the RFC 6761 suffixes `.test`/`.example`/`.invalid`/`.localhost`.
- **Phone:** skip the NANP fictional reserved range `555-0100..555-0199`, robust to `+1`/parens/separators.

For the GitHub-ID case (which needs surrounding context), a skip in the scan loop: a **bare-digit** 10-digit run that matches the phone shape is skipped only when a **GitHub-structured token** (`comment id`, `issuecomment`, `discussion_r`, `/comments/`, `/commit/`, `/pull/`, `/issues/`) sits within 30 chars before it. Bare English words like "commit"/"comment" are intentionally NOT matched, so a real phone near those words in prose is still flagged.

**Not weakened:** real emails on real domains, real phones (including bare-digit phones with no GitHub token), and the boundary cases (`555-0099`, `555-0200`, `notexample.com`) all stay flagged. The reserved values added are standards-mandated unregisterable/fictional, so they can never alias real PII.

## Scope

Primary area: scorer / artifacts PII scanner.

Why this belongs in this repo: machine change; every printed CLI's pii-audit/publish/polish flow inherits the allowlist.

## Catalog Justification

N/A

## Risk

Low and safe-directional: every allowlist errs toward over-flagging (the boundary/lookalike cases stay flagged); the only values skipped are standards-reserved or GitHub-token-adjacent bare IDs. The narrowest residual is the GitHub-ID skip, which is gated tightly (bare digits + structured token) and scoped to markdown/proof prose.

## Output Contract

N/A for generated output (the change is to the scanner, not generate output — no golden impact; `golden verify` passes 24/24). Behavior contract: reserved synthetic emails/phones and GitHub-structured reference IDs are no longer reported/redacted as PII.

## Verification

- `go test ./...` — pass. Includes new table-driven positive/negative/boundary cases in `pii_test.go` (verified failing pre-change), and migration of pre-existing detection/redaction/gate fixtures off `@example.com` (which is now correctly allowlisted) onto a non-reserved domain so they still exercise real-PII handling.
- `scripts/golden.sh verify` — 24/24 pass
- `gofmt` / `golangci-lint run ./internal/artifacts/...` — clean

## AI / Automation Disclosure

- [ ] No AI or automation was used
- [ ] Human-reviewed: AI or automation was used, and a human reviewed the work for intent, fit, and obvious issues before submission
- [x] AI-reviewed only: an AI agent reviewed the work, but no human reviewed it before submission
- [ ] Fully automated: generated and submitted without human review for this specific change

### CODE REVIEW

File: internal/artifacts/pii.go [hunk 1]
```diff
@@ -728,6 +813,9 @@ func scanPIIFileWithRel(path, relSlash string) ([]PIIFinding, error) {
 				if isSyntheticPIIPlaceholder(det.kind, matchedSpan) {
 					continue
 				}
+				if det.kind == PIIKindPhoneUS && isGitHubContextBarePhoneID(line, match[0], matchedSpan) {
+					continue
+				}
```
[the remark this answers is not in the document]
Comment (vinnyp) [reply internal/artifacts/pii.go:820]: Good catch — fixed the drift rather than just the description. `redactPIIPatterns` now routes phone matches through the same `isGitHubContextBarePhoneID` check (refactored to an index-aware replace so it can see the preceding context, which ReplaceAllStringFunc couldn't). A bare GitHub reference ID in a GitHub context is now left intact by redaction too, matching the scan gate; real phones are still redacted. Added `TestRedactPIIText_KeepsGitHubContextIDAndReservedValues`. Pushed in b1f3768c.

**Comment by vinnyp** [comment]
Heads-up: @tmchow had already claimed #1589 (one of the three issues this PR closes) ~6 min before I bundled it here, which I didn't check at the time. Deferring to them on #1589 — they may close this or use it. The other two issues it closes (#1343, #1477) are unclaimed. Flagged on #1589 for a decision.

**Comment by tmchow** [comment]
@mergifyio refresh
